docs/how_to_contribute/math/how_to_add_prime_field.md

```diff
@@ -0,0 +1,409 @@
+# How to add Prime Field
+
+If you want to add new Prime Field, follow this guide.
+
+_Note_: We have our own development conventions. Please read the [conventions docs](/docs/how_to_contribute/conventions.md) before contributing.
+
+## 1. Add `BUILD.bazel` file
+
+Begin by creating a directory named `<prime_field_name>_prime` within `/tachyon/math/finite_field/`, and then add a `BUILD.bazel` file to it. After you have added parameters to your `BUILD.bazel`, Bazel will automatically generate the Prime Field code based on these parameters.
+
+You must choose among `generate_prime_fields()`, `generate_fft_prime_fields()`, or `generate_large_fft_prime_fields()` for code generation, depending on your Prime Field type. For more information, refer to the [`prime_field_generator`](/tachyon/math/finite_fields/generator/prime_field_generator/build_defs.bzl).
+
+Additionally, you can optimize using polygon_zkevm_backend. To utilize this, include `<field_type>_fail.h` and `<field_type>_fail.cc`.
```

Comment:
Maybe you should show more distinctly how using/not using polygon_zkevm_backend works, since it seems optional, and the marsenne-31 example shows only with it.